main.go

```diff
@@ -66,9 +66,11 @@ func main() {
 	internal.Add(http.MethodGet, "/metadata", localCache.Metadata)
 	internal.Add(http.MethodGet, "/digests", localCache.LayerDigests)
 
-	router := e.Group("/v2/:username/:imagename")
+	v2Router := e.Group("/v2")
+	router := v2Router.Group("/:username/:imagename")
```

Comment:
@guacamole lets call router -> `nsRouter` for namespace router?

Reply:
you got it!